### PR TITLE
Notifications fixes

### LINKED ISSUE #11174: Two reply buttons in notifications, none working!
**Describe the bug**

There are now two `reply` button in the notification and neither are working

**To Reproduce**

1. Go to `Notifications` through the bell in the top right
2. Find comment targeted to you
3. Two `reply` buttons are visible 
4. Click on either and nothing happens

**Expected behavior**

There should only be one button and clicking on reply opens up the reply box.

**Screenshots**

Screenshot 2020-10-30 at 12 26 06 PM

What happens in network on click:

Screenshot 2020-10-30 at 12 30 30 PM

**Video**

Oct-30-2020 12-20-34

**Desktop (please complete the following information):**

- OS, version: MacOS Catalina
- Browser, version: Firefox, & Chrome 

**Additional context**

Tried on Firefox, Firefox Developer Edition and Chrome and all had the issue.

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

With recently released Notifications view redesign, I've obviously introduced some bugs and this PR is supposed to address these bugs.

## Related Tickets & Documents

- Closes #11174

## QA Instructions, Screenshots, Recordings

- I'm not sure if the two reply buttons  issue can be tested locally since my guess is it was related to cache and happened only on prod.
- Rest of the stuff in this PR is mostly cosmetics around responsiveness.

## Added tests?

- [ ] Yes
- [x] No, not needed
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/assets/stylesheets/components/avatars.scss [hunk 1]
```diff
@@ -61,6 +61,102 @@
       height: var(--su-10);
     }
   }
+
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/components/avatars.scss:64]: This addition will help us use "responsive" avatars - so you can control the size of the avatar independently from breakpoints (I've used it for Notifications avatars).

File: app/views/notifications/shared/_comment_box.html.erb [hunk 2]
```diff
@@ -68,20 +68,18 @@
           </a>
         </div>
 
-        <% cache "comment-box-comment-reply-#{@last_user_comment}-#{json_data['comment']['updated_at']}-#{json_data['comment']['id']}" do %>
```
Comment (mstruve) [inline app/views/notifications/shared/_comment_box.html.erb:71]: Any idea why we had this cache here in the first place? Does Github history give us any hints?

**Review by mstruve — COMMENTED** [review]
Works well! I would like to know why the cache was there in the first place before we toss it to make sure we are not introducing any performance issues.

File: app/views/notifications/shared/_comment_box.html.erb (hunk 2 above)
Comment (ludwiczakpawel) [reply app/views/notifications/shared/_comment_box.html.erb:71]: It was basically me not really understanding how cache works... With my notifications update yesterday I move some cheese around and I though that comment form must be inside that cache and I simply duplicated existing one...

https://github.com/forem/forem/pull/11084/files#diff-c277bcefb301cd08fba69e9c0056badbff08deb289fe2b4651020f64317c9820L37-L63

^ as you can see the form was never part of the cache and we've never had two same cache boxes (?) here.

Comment (mstruve) [reply app/views/notifications/shared/_comment_box.html.erb:71]: Oooooh I see, you added a second cache where we didn't have one before 👍  mystery solved!

**Review by mstruve — APPROVED** [review]

**Comment by benhalpern** [comment]
> I would like to know why the cache was there in the first place

@mstruve Caching in this area is mostly to mitigate n+1 because it's hard to eager load everything due to the different types of data. But it is sort of a case-by-case basis of how much that would affect things. Some of it _may_ be very necessary, and some not so much. But we don't load in _that many_ notifications in the initial request so it's probably safe to experiment here, and eye any necessary re-implementations.

**Review by benhalpern — APPROVED** [review]

**Pull request merged**